proof_of_concept/asset.py

```diff
@@ -0,0 +1,51 @@
+"""Classes for describing assets."""
+from typing import Any, Dict
+
+
+class Asset:
+    """Asset, a representation of a computation or piece of data."""
+
+    def __init__(self, name: str, data: Any, metadata: Any):
+        """Constructor.
+
+        TODO: typehint Metadata class for metadata argument, problematic
```

Comment:
Fixing this would require quite some refactoring, would maybe be nice to do together, or ignore it for now.

Reply:
Come to think of it, maybe an IMetadata interface would be the simplest fix? I don't like interfaces with only one implementation, but there are a few other instances of this here already anyway, and for some reason this whole thing really wants to be Javaesque anyway, so...